File: coding/24th December 2025/Apple Redistribution into Boxes.java
//Apple Redistribution into Boxes
class Solution 
{
    public int minimumBoxes(int[] apple, int[] capacity) 
    {
        int total = 0;
        for(int ele : apple)
            total += ele;
        Arrays.sort(capacity);
        int result = 0;
        for(int i = capacity.length - 1; i >= 0; i--)
        {
            if(total <= 0)
                break;
            result++;
            total -= capacity[i];
        }
        return result;
    }
}

File: Coding in Python/24th December 2025/Apple Redistribution into Boxes.py
#Apple Redistribution into Boxes
class Solution:
    def minimumBoxes(self, apple: List[int], capacity: List[int]) -> int:
        total = 0
        res = 0
        for ele in apple:
            total += ele
        capacity.sort()
        for i in range(len(capacity) - 1, -1, -1):
            if total <= 0:
                break
            res += 1
            total -= capacity[i]
        return res
